.github/workflows/html5lib-tests.yml

```
@@ -0,0 +1,195 @@
+name: HTML API html5lib Tests
+
+on:
+  push:
+    branches:
+      - trunk
+      # The html5lib tests were introduced in WordPress 6.6.
+      - '6.[6-9]'
+      - '[7-9].[0-9]'
+      - '[1-9][0-9]+.[0-9]'
+    tags:
+      - '6.[6-9]'
+      - '6.[6-9].[0-9]+'
+      - '[7-9].[0-9]'
+      - '[7-9].[0-9].[0-9]+'
+      - '[1-9][0-9]+.[0-9]'
+      - '[1-9][0-9]+.[0-9].[0-9]+'
+  pull_request:
+    branches:
+      - trunk
+      - '6.[6-9]'
+      - '[7-9].[0-9]'
+      - '[1-9][0-9]+.[0-9]'
+    paths:
+      # Any change to a PHP file in the HTML API should run checks.
+      - 'src/wp-includes/html-api/**.php'
+      - 'tests/phpunit/tests/html-api/wpHtmlProcessorHtml5lib.php'
+      - 'tests/phpunit/data/html5lib-tests/**.dat'
```

Changes to this file should also trigger it.

```suggestion
      - 'tests/phpunit/data/html5lib-tests/**.dat'
      # Changes to this file should run the tests to ensure they still run correctly.
      - '.github/workflows/html5lib-tests.yml'
```